hudi-utilities/src/main/java/org/apache/hudi/utilities/deltastreamer/HoodieDeltaStreamer.java

```diff
@@ -416,10 +425,12 @@ public DeltaSync getDeltaSync() {
           jssc.setLocalProperty("spark.scheduler.pool", SchedulerConfGenerator.DELTASYNC_POOL_NAME);
         }
         try {
+          int iteration = 1;
           while (!isShutdownRequested()) {
             try {
               long start = System.currentTimeMillis();
-              Option<String> scheduledCompactionInstant = deltaSync.syncOnce();
+              HoodieMetrics.setTableName(cfg.metricsTableName + "_" + iteration);
```

Comment:
sorry, I don't quite get why we need to set table name here? Next line will in turn rely on the arg passed for tablename. So, don't really understand why we need static fix (i.e. setTableName). From the diff, I see that cfg.tableName is set passed into  DeltaSync.syncOnce(tblName) and HoodieDeltaStreamerMetrics(HoodieWriteConfig tableName). Can you help me understand the case where the static set method for table name is required.

Reply:
IllegalArgumentException is happening because the metrics name are generated in same way using tableName in each run. So we need some way of differentiating metrics names for every run and the easiest way to do that is altering the table name like <tableName>_<iteration>. We need to do this change at 2 places, for HoodieDeltaStreamerMetrics and for HoodieMetrics. 
The table name getting passed with syncOnce() method takes care of HoodieDeltaStreamerMetrics only. For all the other metrics, we need to reset table name in HoodieMetrics class.

Reply:
@pratyakshsharma Thanks for the fix, I have a small question, when using Spark Streaming writing to Hudi (which seems like the continuous mode of deltastreamer), the exception will happen again?

Reply:
@leesf yes. This PR only fixes this for continuous mode of HoodieDeltaStreamer. If you can point me to relevant code of spark datasource from where this can be executed, I can try fixing there as well.

Reply:
@pratyakshsharma just a raw idea, how about adding a static variable (iterator) in HoodieMetrics, and  ++iterator after `updateCommitMetrics`. And back to the PR, if we only fix the HoodieDeltaStreamer, I think we would simply pass the iteration_time to `syncOnce` method to create a new metric name instead of adding a static tableName to HoodieMetrics, I think it is a little bit weird.